### PR TITLE
Add default_service_account query Closes #513

### LINKED ISSUE #513: Add default_service_account query for Kubernetes
### Platform
Kubernetes

### Description
Add a new "Default Service Account", that checks if default service accounts are actively used

### PR DESCRIPTION
**Pull request by rafaela-soares**
Since we cant to ensure that default service accounts are not actively used, it is necessary to check if attribute 'automountServiceAccountToken' is false.  For this query, I thought in two cases:

- Attribute 'automountServiceAccountToken' is undefined

- Attribute 'automountServiceAccountToken' is true

Closes #513

### CODE REVIEW

File: assets/queries/k8s/default_service_account/query.rego [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package Cx
+
+CxPolicy [ result ] {
+    document := input.document[i]
+    
+    document.kind == "ServiceAccount"
+    
+    document.metadata.name == "default"
```
Comment (joelsou5a) [inline assets/queries/k8s/default_service_account/query.rego:8]: Improve searchKey with metadata

metadata := document.metadata
metadata.name == "default"

File: assets/queries/k8s/default_service_account/query.rego [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    
+    object.get(document.metadata, "automountServiceAccountToken", "undefined") == "undefined"
+
+	result := {
+                "documentId": 		  input.document[i].id,
+                "issueType":		  "MissingAttribute",
+                "searchKey": 	      "metadata",
```
Comment (joelsou5a) [inline assets/queries/k8s/default_service_account/query.rego:15]: "searchKey": 	      sprinft("metadata.name=%s", [metadata.name]),

File: assets/queries/k8s/default_service_account/query.rego [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+
+
+CxPolicy [ result ] {
+    document := input.document[i]
+    
+    document.kind == "ServiceAccount"
+    
+    document.metadata.name == "default"
```
Comment (joelsou5a) [inline assets/queries/k8s/default_service_account/query.rego:28]: Improve searchKey with metadata

metadata := document.metadata
metadata.name == "default"

File: assets/queries/k8s/default_service_account/query.rego [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    
+    document.metadata.automountServiceAccountToken != false
+
+	result := {
+                "documentId": 		  input.document[i].id,
+                "issueType":		  "IncorrectValue",
+                "searchKey": 	      "metadata.automountServiceAccountToken",
```
Comment (joelsou5a) [inline assets/queries/k8s/default_service_account/query.rego:35]: "searchKey": sprinft("metadata.name=%s.automountServiceAccountToken", [metadata.name]),

**Review by joelsou5a — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/default_service_account/query.rego [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    metadata.name == "default"
+    
+    object.get(metadata, "automountServiceAccountToken", "undefined") == "undefined"
+
+	  result := {
+                "documentId": 		    input.document[i].id,
+                "issueType":		      "MissingAttribute",
+                "searchKey": 	        sprintf("metadata.name=%s", [metadata.name]),
+                "keyExpectedValue":   sprintf("Attribute 'automountServiceAccountToken' is set in document[%d]",[i]),
+                "keyActualValue": 	  sprintf("Attribute 'automountServiceAccountToken' is undefined in document[%d]",[i])
```
Comment (joaomartinscx) [inline assets/queries/k8s/default_service_account/query.rego:18]: Specify the path where the attribute should be

File: assets/queries/k8s/default_service_account/query.rego [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    metadata.name == "default" 
+    
+    metadata.automountServiceAccountToken == true
+	   
+    result := {
+                "documentId": 		    input.document[i].id,
+                "issueType":		      "IncorrectValue",
+                "searchKey": 	        sprintf("metadata.name=%s.automountServiceAccountToken", [metadata.name]),
+                "keyExpectedValue":   sprintf("Attribute 'automountServiceAccountToken' is false in document[%d]",[i]),
+                "keyActualValue": 	  sprintf("Attribute 'automountServiceAccountToken' is true in document[%d]",[i])
```
Comment (joaomartinscx) [inline assets/queries/k8s/default_service_account/query.rego:38]: Same thing here

File: assets/queries/k8s/default_service_account/metadata.json [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "default_service_account",
+  "queryName": "Default Service Account",
```
Comment (joaomartinscx) [inline assets/queries/k8s/default_service_account/metadata.json:3]: Query name should identify the problem with the resource, in this case, that default service accounts are being used.

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/default_service_account_in_use/query.rego [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    metadata.name == "default" 
+    
+    metadata.automountServiceAccountToken == true
+	   
+    result := {
+                "documentId": 		    input.document[i].id,
+                "issueType":		      "IncorrectValue",
+                "searchKey": 	        sprintf("metadata.name=%s.automountServiceAccountToken", [metadata.name]),
+                "keyExpectedValue":   sprintf("metadata.name=%s has automountServiceAccountToken set a false", [metadata.name]),
+                "keyActualValue": 	  sprintf("metadata.name=%s has automountServiceAccountToken set a true", [metadata.name])
```
Comment (joaomartinscx) [inline assets/queries/k8s/default_service_account_in_use/query.rego:38]: *set to

File: assets/queries/k8s/default_service_account_in_use/query.rego [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+CxPolicy [ result ] {
+    document := input.document[i]
+    
+    document.kind == "ServiceAccount"
+    
+    metadata := document.metadata
+    metadata.name == "default"
+    
+    object.get(metadata, "automountServiceAccountToken", "undefined") == "undefined"
```
Comment (joaomartinscx) [inline assets/queries/k8s/default_service_account_in_use/query.rego:11]: From what I saw in the documentation, *automountServiceAccountToken" is at the resource level, not metadata. Check the first example in this [section](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server), but also try to find other examples, as this the only one I could find.

**Review by joaomartinscx — DISMISSED** [review]

File: assets/queries/k8s/default_service_account_in_use/query.rego (hunk 9 above)
Comment (rafaela-soares) [reply assets/queries/k8s/default_service_account_in_use/query.rego:11]: https://docs.bridgecrew.io/docs/bc_k8s_38

Comment (joaomartinscx) [reply assets/queries/k8s/default_service_account_in_use/query.rego]: https://www.cyberark.com/resources/threat-research-blog/securing-kubernetes-clusters-by-eliminating-risky-permissions#prevent-service-account-token-automounting-on-pods, I found another example, I think it's correct to leave out of metadata

**Review by joaomartinscx — COMMENTED** [review]
Everything looks good to me, I recommend approval and merge of this PR.

**Review by joelsou5a — APPROVED** [review]

**Pull request merged**